qc-runner/src/init.r

```diff
@@ -167,20 +168,39 @@ send_plot_data_to_s3 <- function(pipeline_config, experiment_id, output) {
     return(plot_data_keys)
 }
 
+upload_matrix_to_s3 <- function(pipeline_config, experiment_id, data) {
+
+    s3 <- paws::s3(config=pipeline_config$aws_config)
+
+    object_key <- paste0(experiment_id, '/r.rds')
+
+    count_matrix <- tempfile()
+    saveRDS(data, file=count_matrix)
+
+    message("Uploading updated count matrix to S3 bucket ", pipeline_config$processed_bucket, " at key ", object_key, "...")
+    s3$put_object(
+        Bucket = pipeline_config$processed_bucket,
+        Key = object_key,
+        Body = count_matrix
+    )
+
+    return(object_key)
+
+}
+
 
 wrapper <- function(input_json) {
     # Get data from state machine input.
     input <- RJSONIO::fromJSON(input_json, simplify = FALSE)
-
+    
     str(input)
 
-    c(
-        experiment_id = experimentId,
-        task_name = taskName,
-        config = config,
-        server = server,
-        sample_id = sampleUuid
-    ) %<-% input
+    experiment_id = input$experimentId
+    task_name = input$taskName
+    config = input$config
+    server = input$server
+    upload_count_matrix = input$uploadCountMatrix
+    sample_id = input$sampleUuid
```

Comment:
Nice move, specially when you are dealing with non-R programmers doing R! :-)

Reply:
The syntax was so nice, so I thought it does what it looks like, but things doesn't seem right and found out it doesn't do what it looks like.